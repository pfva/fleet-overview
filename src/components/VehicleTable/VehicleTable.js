import { useContext } from 'react';
import VehicleContext from '../../store/vehicle-context';
import VehicleTableHeader from './VehicleTableHeader/VehicleTableHeader';
import VehicleTableRow from './VehicleTableRow/VehicleTableRow';
import styles from './VehicleTable.module.css';

const VehicleTable = () => {
  const vehicleContext = useContext(VehicleContext);

  return (
    <div className={styles.vehicleTable}>
      <VehicleTableHeader />
      <div className={styles.vehicleTableRowWrapper}>
        {vehicleContext.vehicles.map(vehicle => {
          return <VehicleTableRow key={vehicle.id} vehicle={vehicle} />;
        })}
      </div>
    </div>
  );
};

export default VehicleTable;
