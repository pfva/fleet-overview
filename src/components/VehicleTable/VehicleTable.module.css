.vehicleTable {
  background-color: var(--color-grey-one);
  padding: var(--spacing-normal);
}

.vehicleTableRowWrapper {
  animation: appear-in 1s ease-out forwards;
}

@keyframes appear-in {
  from {
    opacity: 0;
    transform: translateY(3rem);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
}