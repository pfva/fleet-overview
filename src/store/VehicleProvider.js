import { useState } from 'react';
import VehicleContext from '../store/vehicle-context';
import vehiclesJson from '../data/vehicles.json';

const VehicleProvider = props => {
  const [vehicleState, setVehicleState] = useState(vehiclesJson);

  const updateVehicleHandler = (id, updatedVehicle) => {
    const vehicleToUpdateIndex = vehicleContext.vehicles.findIndex(vehicle => vehicle.id === id);
    vehicleContext.vehicles[vehicleToUpdateIndex] = updatedVehicle;
    
    setVehicleState(vehicleContext.vehicles);
  }

  const vehicleContext = {
    vehicles: vehicleState,
    updateVehicle: updateVehicleHandler,
  };

  return (
    <VehicleContext.Provider value={vehicleContext}>
      {props.children}
    </VehicleContext.Provider>
  );
};

export default VehicleProvider;
