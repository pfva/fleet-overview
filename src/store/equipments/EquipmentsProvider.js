import { useState } from 'react';
import EquipmentsContext from './equipments-context';
import equipmentsJson from '../../data/equipments.json';

const EquipmentsProvider = props => {
  const [equipmentsState, setEquipmentsState] = useState(equipmentsJson);

  const equipmentsContext = {
    equipments: equipmentsState,
  };

  return (
    <EquipmentsContext.Provider value={equipmentsContext}>
      {props.children}
    </EquipmentsContext.Provider>
  );
};

export default EquipmentsProvider;
