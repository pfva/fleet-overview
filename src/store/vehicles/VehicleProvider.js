import { useState } from 'react';
import VehicleContext from './vehicle-context';
import vehiclesJson from '../../data/vehicles.json';

const VehicleProvider = props => {
  const [vehicleState, setVehicleState] = useState(vehiclesJson);

  const updateVehicleHandler = (id, updatedVehicle) => {
    const vehicleToUpdateIndex = vehicleState.findIndex(
      vehicle => vehicle.id === id
    );
    const updatedVehicleState = [...vehicleState];
    updatedVehicleState[vehicleToUpdateIndex] = updatedVehicle;
    setVehicleState(updatedVehicleState);
  };

  const vehicleContext = {
    vehicles: vehicleState,
    updateVehicle: updateVehicleHandler,
  };

  return (
    <VehicleContext.Provider value={vehicleContext}>
      {props.children}
    </VehicleContext.Provider>
  );
};

export default VehicleProvider;
