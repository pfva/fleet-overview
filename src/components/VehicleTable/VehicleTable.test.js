import renderer from 'react-test-renderer';
import VehicleTable from './VehicleTable';
import VehicleProvider from '../../store/VehicleProvider';
import EquipmentsProvider from '../../store/EquipmentsProvider';

describe('VehicleTable', () => {
  it('renders an array of vehicles', () => {
    const tree = renderer
      .create(
        <VehicleProvider>
          <EquipmentsProvider>
            <VehicleTable />
          </EquipmentsProvider>
        </VehicleProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
