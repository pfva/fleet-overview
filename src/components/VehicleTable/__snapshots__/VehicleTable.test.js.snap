// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleTable renders an array of vehicles 1`] = `
<div
  className="vehicleTable"
>
  <div
    className="vehicleTableHeader"
  >
    <p
      className="vehicleHeaderCell"
    >
      ID
    </p>
    <p
      className="vehicleHeaderCell"
    >
      Name
    </p>
    <p
      className="vehicleHeaderCell"
    >
      Driver
    </p>
    <p
      className="vehicleHeaderCell"
    >
      Status
    </p>
    <p
      className="vehicleHeaderCell"
    >
      Fuel type
    </p>
    <p
      className="vehicleHeaderCell"
    >
      Equipments
    </p>
  </div>
  <div
    className="vehicleTableRowWrapper"
  >
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v1
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          AB23
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          SpongeBob SquarePants
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LNG
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Crane
          </p>
          <p
            className="vehicleTableCell"
          >
            Tachograph
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v2
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          XXW123
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Patrick Star
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Tachograph
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v3
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          GA33
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Squidward Tentacles
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          CNG
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Hook
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v4
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LK21
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Mr. Krabs
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          inactive
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        />
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v5
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          GHB 5
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Plankton and Karen
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Crane
          </p>
          <p
            className="vehicleTableCell"
          >
            Hook
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v6
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LLO21
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Sandy Cheeks
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Fire Extinguisher
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v7
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          OWL21
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Mrs. Puff
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Electrical
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        />
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v8
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          JNWJ12
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Pearl Krabs
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Eletrical
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Fire Extinguisher
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v9
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          KL2OK1
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Gary the Snail
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LNG
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        />
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v10
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LSAO21
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Patchy the Pirate
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LNG
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Tachograph
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v11
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          LSP21P
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Potty the Parrot
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          inactive
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Hook
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v12
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          212AWN
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          French Narrator
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        />
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v13
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          1W321W
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Mermaid Man and Barnacle Boy
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Tachograph
          </p>
        </span>
      </div>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v14
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          14HUEO
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          The Flying Dutchman
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        />
      </p>
    </div>
    <div
      className="vehicleTableRow"
    >
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          v15
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          
        </span>
      </p>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          King Neptune
        </span>
      </p>
      <div
        className="vehicleTableCell iconWrapper"
        onClick={[Function]}
      >
        <span
          className="dot"
          data-testid="Icon-active"
        />
        <p>
          <span
            className="dataText"
          >
            active
          </span>
        </p>
      </div>
      <p
        className="vehicleTableCell"
        onClick={[Function]}
      >
        <span
          className="dataText"
        >
          Diesel
        </span>
      </p>
      <div
        className="vehicleEquipments"
        data-testid="VehicleEquipments-wrapper"
        onClick={[Function]}
      >
        <span
          className="dataText adjustHovered"
        >
          <p
            className="vehicleTableCell"
          >
            Crane
          </p>
          <p
            className="vehicleTableCell"
          >
            Tachograph
          </p>
          <p
            className="vehicleTableCell"
          >
            Fire Extinguisher
          </p>
          <p
            className="vehicleTableCell"
          >
            Hook
          </p>
        </span>
      </div>
    </div>
  </div>
</div>
`;
